frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.node-selector.scenario.ts

```diff
@@ -0,0 +1,52 @@
+import { browser, ExpectedConditions as until } from 'protractor';
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import { createResource, deleteResource, click } from '@console/shared/src/test-utils/utils';
+import * as editNodeSelectorView from '../views/editNodeSelectorView';
+import * as virtualMachineView from '../views/virtualMachine.view';
+import { VM_CREATE_AND_EDIT_TIMEOUT_SECS } from './utils/consts';
+import { VirtualMachine } from './models/virtualMachine';
+import { getVMManifest } from './utils/mocks';
+import { getRandStr } from './utils/utils';
+
+describe('KubeVirt VM detail - edit Node Selector', () => {
+  const testVM = getVMManifest('Container', testName, `node-selector-vm-${getRandStr(5)}`);
+  const vm = new VirtualMachine(testVM.metadata);
+
+  beforeAll(async () => {
+    createResource(testVM);
+  });
+
+  afterAll(() => {
+    deleteResource(testVM);
+  });
+
```

Comment:
I think it would be worth adding tests that the the dialog displays found nodes for existing label and displays warning for non-existing label.

Reply:
hmm..how do we know which nodes will be present when test is tirggered?

Reply:
We could query worker nodes, get labels of an arbitrary node and pick an arbitrary label of that node. Then we could use that label knowing that there is at least one node satisfying the condition.

I'm totally okay with adding this later on as a followup.